hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/MappingJdbcSqlConnectorTest.java

```diff
@@ -311,6 +312,33 @@ public void when_mappingIsDeclaredWithDataConnection_then_itIsAvailable() throws
         assertThat(object).isEqualTo(name);
     }
 
+    /**
+     * Source : <a href=https://github.com/hazelcast/hazelcast/issues/24337">issue #24337</a>.
+     */
+    @Test
+    public void given_mappingIsDeclaredWithDataConn_when_DataConnWasRemoved_then_success() throws Exception {
+        // given
+        String dlName = randomName();
+        String name = randomName();
+        createTable(name);
+        Map<String, String> options = new HashMap<>();
+        options.put("jdbcUrl", dbConnectionUrl);
+
+        // when
+        createDataConnection(instance(), dlName, "JDBC", false, options);
+        createJdbcMappingUsingDataConnection(name, dlName);
+        sqlService.execute("DROP DATA CONNECTION " + dlName);
+
+        // then
+        assertThat(dlName).isNotEqualTo(TEST_DATABASE_REF);
+        assertRowsAnyOrder("SHOW DATA CONNECTIONS ", singletonList(new Row(TEST_DATABASE_REF)));
+
+        // Mapping doesn't provide data, since data connection was removed.
+        assertThatThrownBy(() -> sqlService.execute("SELECT * FROM " + name))
```

Comment:
I think you should execute an unrelated SQL query or `SHOW MAPPINGS` to ensure that everything works and fully reproduce the error